overlord/snapstate/snapmgr.go

```diff
@@ -420,6 +431,10 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 			newSeq = append(newSeq, si)
 		}
 		snapst.Sequence = newSeq
+		// update Current if it just got removed
+		if snapst.Current == ss.Revision {
```

Comment:
do we really need this ? it worries me, we shouldn't discard current like that, it should have been moved already